File: solutions/cube-root/cube-root.js
// This solution is based on the Newton-Raphson Method

var cubeRoot = function (number) {
  var guess = 1;
  var accuracy = 0.0001; // The accuracy of your response
  var improve = function (y) {
    return ((number / Math.pow(y, 2)) + (2 * y)) / 3;
  };
  var loop = function () {
    if (!(Math.abs(number - Math.pow(guess, 3)) < accuracy)) {
      guess = improve(guess);
      return loop();
    } else {
      return guess;
    };
  };
  return loop();
};

File: solutions/cube-root/cube_root.py
def improve(n, guess):
    return (n / guess**2 + 2 * guess) / 3


def cub_root(n):
    accuracy = 0.01
    result = improve(n, 1)
    while abs(n - result**3) > accuracy:
        result = improve(n, result)
    return result
